lib/puppet/provider/image_streamer_golden_image/image_streamer.rb

```diff
@@ -21,32 +21,29 @@
 
   mk_resource_methods
 
-  def exists?
-    super([nil, :found, :download_details_archive, :download])
```

Comment:
both `:download_details_archive` and `:download` ensure methods invoke the `get_single_resource_instance` method, which means they require at least some info passed inside `data` to be able to retrieve a resource.
Invoking `super` as it was allowed for the `data` field to be passed in empty ({}) or not passed in at all for these ensure methods, which should not be happening.
Removing those two methods from the `super` declaration, we'd get `super([nil, :found])`, which is already default, so there's no need to call it.